src/Integration.Vsix/Analysis/DocumentEventsHandler.cs

```diff
@@ -73,6 +91,11 @@ private void OnOpenDocumentRenamed(object sender, DocumentRenamedEventArgs args)
         {
             vcxCompilationDatabaseUpdater.RenameFileAsync(args.OldFilePath, args.Document.FullPath).Forget();
         }
+        threadHandling.RunOnBackgroundThread(() =>
```

Comment:
One of the reasons I wanted to not do thread switching in `vcxCompilationDatabaseUpdater` and do it in the callers was because they will mort likely need to do that anyways. Here I would prefer if `vcxCompilationDatabaseUpdater` was also called inside `RunOnBackgroundThread` for consistency